### PR TITLE
Feature/water sand grass beaches WIP OH GOD WHAT HAVE I DONE

### PR DESCRIPTION
**Pull request by mortizzle**

### CODE REVIEW

File: src/Trains.NET.Rendering.Skia/SKCanvasWrapper.cs [hunk 1]
```diff
@@ -67,6 +68,47 @@ public void GradientRect(float x, float y, float width, float height, Color star
             _canvas.DrawRect(x, y, width, height, paint);
         }
 
+        public void GradientRectLeftRight(float x, float y, float width, float height, IEnumerable<Color> colours)
```
Comment (mortizzle) [inline src/Trains.NET.Rendering.Skia/SKCanvasWrapper.cs:71]: I think we should probably consolidate all of these and add some kind of gradientdirection flag instead

File: src/Trains.NET.Rendering/LayerRenderer/TerrainTypeRenderer.cs [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+                var gradient = GenerateGradientForCornerCell(terrain, adjacency, orthogonalGradients);
+                if (gradient.GradientType == GradientType.None) continue;
+                cornerGradients[adjacency] = gradient;
+            }
+
+            RenderBackground(terrain, canvas);
+
+            // Render edge gradients, highest weight last
+            foreach (var gradient in orthogonalGradients.Values.Where(g => g.GradientType != GradientType.None).OrderBy(g => GetWeightForGradientType(g.GradientType))) {
+                _gradientRenderer.Render(canvas, gradient);
```
Comment (mortizzle) [inline src/Trains.NET.Rendering/LayerRenderer/TerrainTypeRenderer.cs:66]: Maybe EdgeGradientRenderer and CornerGradientRenderer as distinct services may make more sense

**Comment by mortizzle** [comment]
I think I need to wrap this in some tests and then refactor it so that it's less gross

**Comment by ScottHolden** [comment]
^^^ and the conflicts :P

**Pull request closed without merging**